assets/php/class-wgpb-block-product-new.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Display the Newest Products block in the post content.
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-gutenberg-products-block.
+ *
+ * @package WooCommerce\Blocks
+ * @version 2.1.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Handler for getting newest products for display.
+ */
+class WGPB_Block_Product_New extends WGPB_Block_Grid_Base {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'product-new';
+
+	/**
+	 * Set args specific to this block
+	 *
+	 * @param array $query_args Query args.
+	 */
+	protected function set_block_query_args( &$query_args ) {
+		$query_args['orderby'] = 'date';
+		$query_args['order']   = 'DESC';
+	}
+}
+
+/**
+ * Render the Newest Products block.
+ *
+ * @param array  $attributes Block attributes. Default empty array.
+ * @param string $content    Block content. Default empty string.
+ * @return string Rendered block type output.
+ */
+function wgpb_block_render_product_new( $attributes, $content ) {
```

Comment:
I don't think we should mix functions and class files. This function should be put elsewhere - it could even do the include so that `WGPB_Block_Product_New` class is only included when needed.

Reply:
I'm not entirely sure what you mean- if we merge this as-is, maybe you can add a PR to do that?

Reply:
`class-` files should ideally only contain the PHP class, not functions or other code. My suggestion was to move `wgpb_block_render_product_new` function into a different file (maybe a collection of similar functions) and leave only the class definition in this file.

You could then, optionally, move...

```
require_once dirname( __FILE__ ) . '/class-wgpb-block-product-new.php';
```

...inside the `wgpb_block_render_product_new()` function so that the class is only included when `wgpb_block_render_product_new` is called.

Hope that makes more sense.

Reply:
Taken care of this in 3da66033fd98048b7999344c83d5e612567ba718